File: pstu_map/src/main/java/com/project/pstu_map/models/Room.java
package com.project.pstu_map.models;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "rooms")
@Data
public class Room {

    @Id
    private String id;

    @Column(nullable = false)
    private String name;

    private int floor;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "building_id")
    private Building building;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "node_id")
    private Node node;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "room_type_id", nullable = false)
    private RoomType roomType;

    @Column(name = "qr_code", unique = true, length = 100)
    private String qrCode;

    @Column(name = "room_polygon", columnDefinition = "TEXT")
    private String roomPolygon;

    @Column(name = "custom_color", length = 7)
    private String customColor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "custom_icon_id")
    private UploadedIcon customIcon;
}

File: frontend/shared-frontend/types/Room.ts
import type {Node} from "./Node.ts";
import type {RoomType} from "./RoomType";

export interface Room {
    id: string;
    name: string;
    floor: number;
    roomType: RoomType;
    description: string;
    node: Node | null;
    buildingId?: number;
    qrCode?: string;
    roomPolygon?: string;
    customColor?: string;
    customIconPath?: string;
}